x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,718 @@
+<<use-rolling-restarts,rolling restart>>. {es} will not automatically reload changes for
+passwords stored in the secure settings. 
+
+[[use-rolling-restarts]]
+.Rolling restarts are preferred
+****
+While it's possible to do an in-place update for security certificates, using
+a <<restart-cluster-rolling,rolling restart>> on your cluster is safer. An in-place update avoids some
+complications of a rolling restart, but incurs the following risks:
+
+* If you use PEM files, your certificate and key are in separate files. You
+must update both files _simultaneously_ or the node might experience a temporary
+period where it cannot establish new connections.
+* Updating the certificate and key does not automatically force existing
+connections to refresh. This means that even if you make a mistake, a node can
+seem like it's functioning but only because it still has existing connections.
+It's possible that a node will be unable to connect with other nodes, rendering
+it unable to recover from a network outage or node restart.
+****
+
+[[update-node-certs-same]]
+=== Update certificates with the same CA
+++++
+<titleabbrev>With the same CA</titleabbrev>
+++++
+
+This procedure assumes that the you have access to the CA certificate and key
+that was originally generated (or otherwise held by your organization) and used
+to sign the node certificates currently in use. It also assumes that the
+clients connecting to {es} on the HTTP layer are configured to trust the CA
+certificate.
+
+If you have access to the CA used to sign your existing certificates, you only
+need to replace the certificates and keys for each node in your cluster. If you
+replace your existing certificates and keys on each node and use the same
+filenames, {es} reloads the files starts using the new certificates and keys.
+
+You don't have to restart each node, but doing so forces new TLS connections and is <<use-rolling-restarts,a recommended practice>> when updating certificates.
+Therefore, the following steps include a node restart after updating each
+certificate.
+
+The following steps provide instructions for generating new node certificates
+and keys for both the transport layer and the HTTP layer. You might only need
+to replace one of these layer's certificates depending on which of your
+certificates are expiring.
+
+[[cert-password-updates]] 
+IMPORTANT: If your keystore is password protected, the password
+is stored in the {es} secure settings, _and_ the password needs to change, then
+you must perform a <<restart-cluster-rolling,rolling restart>> on your cluster.
+You must also use a different file name for the keystore so that {es} doesn't
+reload the file before the node is restarted.
+
+TIP: If your CA has changed, complete the steps in 
+<<update-node-certs-different,update security certificates with a different CA>>.
+
+[[node-certs-same-transport]]
+==== Generate a new certificate for the transport layer
```

Comment:
we're now making the assumption that `xpack.security.transport.ssl.verificaion_mode: certificate` is set  (which is what we suggest in other places in the doc, but it's _not_ the default value. Should we point it out ?

Reply:
Are you suggesting that we add `xpack.security.transport.ssl.verificaion_mode: certificate` to this [YAML example](https://github.com/elastic/elasticsearch/pull/73781/files/3be58771f839d1b4721aac855ff77a7fd476b7d8#diff-1490c74621f912bbdeb341055e902df727b240257b2fbd22bf774da6f8332f70R98-R104)? I'm +1 to that change, but just making sure I understand correctly. Same change would apply to your comment in the [later example](https://github.com/elastic/elasticsearch/pull/73781#discussion_r671205963).

Reply:
Either that or a hint/note/warning that this set of instructions assumes that this is _already_ the case or a combination of both

Reply:
Cool. I'll add that setting in the yaml file and also a note that we assume that the provided certificate is signed by a trusted CA and the verification mode is set to `certificate`.

Reply:
for future reference, we only need to put this in because we are going by (here and in other part of our docs) the simplest setup possible, which is that all of our nodes share the same key/certificate for transport TLS. The verification mode needs to be set to `certificate` so that nodes won't attempt to do hostname verification